components/raftstore/src/store/fsm/peer.rs

```diff
@@ -2470,6 +2472,43 @@ where
         self.fsm.hibernate_state.count_vote(from.get_id());
     }
 
+    fn on_collect_peer_apply_index_process_request(&mut self, from: &metapb::Peer) {
+        if self.fsm.peer.is_leader() {
+            unreachable!();
+        }
+        let mut msg = ExtraMessage::default();
+        msg.set_type(ExtraMessageType::MsgCollectPeerApplyIndexProgressResponse);
+        // FIXME: add a new field in ExtraMessage or use ctx?
+        msg.premerge_commit = self.fsm.peer.last_applying_idx;
+        self.fsm
+            .peer
+            .send_extra_message(msg, &mut self.ctx.trans, from);
```

Comment:
Send apply index with raft normal messages(heartbeat or append), no need to send it separately.